### DISCUSSION TITLE
Castables don't work when used from a constructor

### DISCUSSION THREAD

**Discussion by savr-reda** [spatie/laravel-data#395] [Q&A] [answered]
Hi everyone, I am using the latest 3.2.1 version of the package with Laravel 10 and PHP 8.2. As an example, I used [the sample provided in the documentation](https://spatie.be/docs/laravel-data/v3/advanced-usage/creating-a-cast#content-castables):

```php
class ForgotPasswordRequest extends Data
    public function __construct(
        #[WithCastable(Email::class)]
        public Email $email,
    ) {}
}
```

```php
class Email implements Castable
{
  public function __construct(public string $email)
  {
  }

  public static function dataCastUsing(...$arguments)
  {
    return new class implements Cast
    {
        public function cast(DataProperty $property, mixed $value, array $context): Email
        {
            return new Email($value);
        }
    };
  }
}
```

When I try to create the request via the constructor, I get a type error message:

```php
$forgotPasswordRequest = new ForgotPasswordRequest('test@email.com');
```
```
TypeError: ForgotPasswordRequest::__construct(): Argument #1 ($email) must be of type Email, string given
```

However, it works as expected when using the `from` method:
```php
$forgotPasswordRequest = ForgotPasswordRequest::from(['email' => 'test@email.com']);
```

Since the documentation is not explicit enough about that part, is that supposed to be an issue or the intended behavior?

**Comment by rubenvanassche** [comment, marked as answer]
This is expected behavior, we cannot overwrite this stuff since it is not possible in PHP

**Comment by savr-reda** [comment]
Hi @rubenvanassche, glad it got confirmed then. Thank you for your answer!
